sdks/python/apache_beam/io/fileio.py

```diff
@@ -624,11 +636,19 @@ def open_writer(self, init_result, uid):
     # We also ensure there will be no collisions with uid and a
     # (possibly unsharded) file_path_prefix and a (possibly empty)
     # file_name_suffix.
+    if not self.file_path_prefix.is_accessible():
+      raise RuntimeError('%s not accessible' % self.file_path_prefix)
+    file_path_prefix = self.file_path_prefix.get()
+    file_name_suffix = self.file_name_suffix.get()
     suffix = (
-        '.' + os.path.basename(self.file_path_prefix) + self.file_name_suffix)
+        '.' + os.path.basename(file_path_prefix) + file_name_suffix)
     return FileSinkWriter(self, os.path.join(init_result, uid) + suffix)
 
   def finalize_write(self, init_result, writer_results):
+    if not self.file_path_prefix.is_accessible():
```

Comment:
@chamikaramj 

Is it enough to check for `is_accessible` only at `initialize_write`?